lib/driver.js

```diff
@@ -292,45 +291,24 @@ class XCUITestDriver extends BaseDriver {
 
     log.info(`Setting up ${this.isRealDevice() ? 'real device' : 'simulator'}`);
 
-    if (this.isRealDevice()) {
-      if (this.opts.app) {
-        await this.installApp();
-        this.logEvent('appInstalled');
-      }
-    } else {
+    if (this.isSimulator()) {
       this.localeConfig = await iosSettings.setLocale(this.opts.device, this.opts, {}, this.isSafari());
       await iosSettings.setPreferences(this.opts.device, this.opts, this.isSafari());
       // Cleanup of installd cache helps to save disk space while running multiple tests
       // without restarting the Simulator: https://github.com/appium/appium/issues/9410
       await this.opts.device.clearCaches('com.apple.mobile.installd.staging');
 
-      let installAppPromise = null;
```

Comment:
Is all of this handled somewhere else? Why is it removed in this PR?

Reply:
the previous logic was using ON_BOOT event handler to init application install. This makes the flow a bit faster, but overcomplicates the code from my point of view. I just want to start logging before application install (simulator should be in `booted` state in order for this logging algorithm to work) and don't want to make the logic even more complicated there.

Reply:
I'm all for removing complexity!